### PR TITLE
Test/TestDirectory: default to `showProgress:=false` if stdout isn't a tty (for stable-4.9)

### LINKED ISSUE #2049: Changed default behaviour of Test
The documented default for the `showProgress` component
of the optional argument record for `Test` was `false` up to GAP 4.8.8.
In the master branch, the documented default is `"some"`.
This means that every personal test setup will have to be changed
in order to set the value explicitly to `false`.
Was this change really intended?

### PR DESCRIPTION
**Pull request by fingolfin**
Thus if you pipe the output of Test or TestDirectory into a file, no progress output is generated.

Note that only the default setting is changed; you can still manually override the `showProgress` setting.

Should resolve #2049 -- curious to hear what @ThomasBreuer thinks.

### CODE REVIEW

**Comment by ThomasBreuer** [comment]
This looks fine.
Thank you for this change.

Once `IS_OUTPUT_TTY` becomes available in a released GAP version,
I will use it for example in some places in the Browse package.

File: lib/test.gi [hunk 1]
```diff
@@ -535,7 +542,8 @@ end);
 ##  <Item>If <K>true</K>, stop as soon as any <Ref Func="Test" /> fails (defaults to <K>false</K>).
 ##  </Item>
 ##  <Mark><C>showProgress</C></Mark>
-##  <Item>Print information about how tests are progressing (defaults to <K>"some"</K>).
+##  <Item>Print information about how tests are progressing (defaults to <K>"some"</K>
```
Comment (alex-konovalov) [inline lib/test.gi]: This is the the place to replace <K> by <C> as @ThomasBreuer mentioned.

**Comment by fingolfin** [comment]
Fixed the GAPDoc issue, thanks for spotting it!

**Comment by alex-konovalov** [comment]
@fingolfin thanks - happy to merge this after next CI tests round.

**Pull request merged**